### DISCUSSION TITLE
Anchor skin/HUD layout elements to playfield

### DISCUSSION THREAD

**Discussion by puyoxyz** [ppy/osu#17076] [Ideas]
My osu!lazer layout looks like this: (with 50%/50%/50%/50% gameplay scaling)

![image](https://user-images.githubusercontent.com/15311104/156539380-8013de74-4a32-404e-8cbf-186184c8d260.png)

It works fine, but if I want to change my playfield size (i.e. if I'm playing a map with very small circles and want to use a bigger playfield instead of my usual small one), my layout/HUD stays in the same place, so I have to either turn it off or manually move everything again.

It would be nice if layout elements could be anchored to the *playfield* (i.e. top left of playfield for my score and hp bar, top right of playfield for the pp counter, bottom right for acc, etc...) so that it adjusted whenever I resize my playfield.

And yes, I know I could just set my scaling to scale everything, but I don't like how it looks (with the background being small and having a separate background around it) + I still want the big song progress bar at the bottom

**Comment by peppy** [comment]
Definitely sounds like something we can offer when expanding the skin editor's functionality.

**Comment by Bruno5430** [comment]
This would fit great for mania.
Something like this could definitely fix #22362
